frontend/packages/kubevirt-plugin/src/components/modals/create-vm-wizard.tsx

```diff
@@ -0,0 +1,96 @@
+import * as _ from 'lodash';
+
+import {
+  CreateVmWizard,
+  getResource,
+  TEMPLATE_TYPE_LABEL,
+  TEMPLATE_TYPE_VM,
+  TEMPLATE_TYPE_BASE,
+} from 'kubevirt-web-ui-components';
+
+import {
+  NamespaceModel,
+  TemplateModel,
+  PersistentVolumeClaimModel,
+  StorageClassModel,
+} from '@console/internal/models';
+import { modalResourceLauncher } from '@console/shared';
+import { units, Firehose } from '@console/internal/components/utils';
+import { k8sCreate, k8sGet, k8sPatch, k8sKill } from '@console/internal/module/k8s';
+import {
+  VirtualMachineModel,
+  NetworkAttachmentDefinitionModel,
+  DataVolumeModel,
+} from '../../models';
+
+export const openCreateVmWizard = (activeNamespace, createTemplate = false) => {
+  const resources = [
+    getResource(NamespaceModel, { prop: 'namespaces', optional: true }),
+    getResource(VirtualMachineModel, { prop: 'virtualMachines' }),
+    getResource(TemplateModel, {
+      namespace: activeNamespace,
+      prop: 'userTemplates',
+      matchLabels: { [TEMPLATE_TYPE_LABEL]: TEMPLATE_TYPE_VM },
+    }),
+    getResource(TemplateModel, {
+      namespace: 'openshift',
+      prop: 'commonTemplates',
+      matchLabels: { [TEMPLATE_TYPE_LABEL]: TEMPLATE_TYPE_BASE },
+    }),
+    getResource(NetworkAttachmentDefinitionModel, {
+      namespace: activeNamespace,
+      prop: 'networkConfigs',
+    }),
+    getResource(StorageClassModel, { prop: 'storageClasses' }),
+    getResource(PersistentVolumeClaimModel, {
+      namespace: activeNamespace,
+      prop: 'persistentVolumeClaims',
+    }),
+    getResource(DataVolumeModel, { namespace: activeNamespace, prop: 'dataVolumes' }),
+  ];
+
+  const resourcesToProps = (props) => {
+    const { namespaces, userTemplates, commonTemplates } = props;
+    const flatten = (resourceName) => ({ [resourceName]: _.get(props[resourceName], 'data') });
+
+    let selectedNamespace;
+    if (activeNamespace) {
+      selectedNamespace = _.get(namespaces, 'data', []).find(
+        (namespace) => namespace.metadata.name === activeNamespace,
+      );
+    }
+
+    let templates;
+    if (userTemplates && commonTemplates) {
+      templates = _.get(userTemplates, 'data', []).concat(_.get(commonTemplates, 'data', []));
+    }
+
+    // namespaces, templates, dataVolumes, virtualMachines
+    return {
+      selectedNamespace,
+      templates,
+
+      ...flatten('namespaces'),
```

Comment:
can we use map here? or reduce..

Reply:
I think recent syntax is much more readable. The list of names would still need to be somewhere so I don't see much benefit it changing that unless there is other strong opinion on that.

Reply:
ok, it is not a must